jsbank/modal.js

```diff
@@ -0,0 +1,126 @@
+import {addList} from './main.js';
+
+/*모달 여닫는 버튼 추가(애니메이션 효과 포함)*/
+const modalWrapper =document.querySelector(".modal_wrapper");
+const modalBtn = document.getElementById("add_btn");
+const closeBtn = document.getElementById("close");
+
+modalBtn.onclick = () => {
+    modalWrapper.style.bottom = "0";
+
+    const backgroundElement = document.createElement("div");
+    backgroundElement.classList.add("dark");
+
+    modalWrapper.before(backgroundElement);
+};
+closeBtn.onclick = () => {
+    modalWrapper.style.bottom = "-40rem";
+
+    document.querySelector(".dark").remove();
+};
+
+/*모달 체크 박스 설정 및 카테고리 연동*/
+const modalInbox = document.getElementById("input_modalbox");
+const modalOutbox = document.getElementById("output_modalbox");
+const modalInput = document.getElementById("select_input");
+const modalOutput = document.getElementById("select_output");
+
+let newCategory = document.getElementById("category_input");
+let tempType = "input";
+
+function checkOnly(e){
+    if(e.target === modalInbox){
+        modalInbox.checked = true;
+        modalOutbox.checked = false;
+
+        modalInput.style.display = "flex";
+        modalOutput.style.display = "none";
+
+        tempType = "input";
+        newCategory = document.getElementById("category_input");
+    }
+    else{
+        modalInbox.checked = false;
+        modalOutbox.checked = true;
+
+        modalInput.style.display = "none";
+        modalOutput.style.display = "flex";
+
+        tempType = "output";
+        newCategory = document.getElementById("category_output");
+    }
+}
+
+modalInbox.addEventListener('click', checkOnly);
+modalOutbox.addEventListener('click', checkOnly);
+
+/*금액 칸에 숫자만 입력 받기*/
+const newAmount = document.getElementById("amount");
+const newTitle = document.getElementById("title");
+
+function onlyNum(event){
+    let val = event.target.value;
+    val = val.replace(/,/g,"");
+
+    if(val.match(/[^0-9]/g)){
+        val = val.replace(/[^0-9]/g,"");
+        alert('숫자만 입력해주세요.');
+    }
+
+    /[0-9]/g.test(event.target.value) === false
+    ? event.target.value = ""
+    :event.target.value = parseInt(val).toLocaleString();
```

Comment:
```suggestion
   const TEXT_EXPRESSION = /[0-9]/g;
    event.target.value = TEXT_EXPRESSION.test(event.target.value)
    ?  parseInt(val).toLocaleString() : "" ;
```
요로코롬 가능하죠! 정규표현식 같은 경우엔 이렇게 상수화해서 빼주는게 좋답니당 ! 추후 유지보수할 때 바뀔 경우 유용하기도 하구요!